### DISCUSSION TITLE
Can PMD be used to print out the cyclomatic complexity of a given class?

### DISCUSSION THREAD

**Discussion by edmondop** [pmd/pmd#3252] [Q&A]
I know there are rulesets which contains cyclomatic complexity, but what about proving you are now under a given threshold, how can I do it to show a developer he has improved a lot?

**Comment by roelWuyts** [comment]
I have a similar question where I would like to use PMD as a metrics engine, just showing me the 'raw metrics' results.

Looking through the documentation and the source code, the best option I found was to run with a treshold of 1 (so it shows the result as a violation no matter what) and then parsing or regexp-ing the result message to retrieve the value. 

Not quite useful for my use case but maybe this helps you ?

This is the rule I was playing with (I was only interested in classes so set that treshold to 1, while setting to some large value so there are no reports on methods):
`<rule ref="category/java/design.xml/CyclomaticComplexity" >
        <priority>1</priority>
        <properties>
            <property name="classReportLevel" value="1" />
            <property name="methodReportLevel" value="9999" />
        </properties>
    </rule>`

Yes, this is really clumsy but then again I am a complete newbie in PMD ;-)
